crates/zk-por-core/src/merkle_sum_prover/circuits/merkle_sum_circuit.rs

```diff
@@ -29,19 +29,12 @@ pub struct MerkleSumNodeTarget {
 }
 
 impl MerkleSumNodeTarget {
-    pub fn default() -> Self {
-        MerkleSumNodeTarget {
-            sum_equity: Target::default(),
-            sum_debt: Target::default(),
-            hash: [Target::default(); NUM_HASH_OUT_ELTS].into(),
-        }
-    }
-
     /// Given children nodes, generate the MerkleSumNodeTarget
     pub fn get_parent_from_children<const N: usize>(
         builder: &mut CircuitBuilder<F, D>,
-        children: [&MerkleSumNodeTarget; N],
+        children: &Vec<MerkleSumNodeTarget>,
```

Comment:
that's the same? what is the reason changing to a vector

Reply:
Because in the old code we go from arr -> vec but if we just use a vec we avoid the conversion